src/app/services/graph-client/MsalAuthProvider.ts

```diff
@@ -0,0 +1,116 @@
+import { AuthenticationProvider } from '@microsoft/microsoft-graph-client';
+import { MSALAuthenticationProvider } from '@microsoft/microsoft-graph-client/lib/src/MSALAuthenticationProvider';
+
+import { DEFAULT_USER_SCOPES } from '../graph-constants';
+
+export class MsalAuthProvider implements AuthenticationProvider {
+  public authProvider: any;
+  public userAgentApp: any;
+  private loginType = this.getLoginType();
+  private defaultUserScopes: string[];
+
+  constructor() {
+    const clientId = process.env.REACT_APP_CLIENT_ID || '';
+    const options = {
+      redirectUri: window.location,
+      storeAuthStateInCookie: true,
+    };
+    this.defaultUserScopes = DEFAULT_USER_SCOPES.split(' ');
+    this.authProvider = new MSALAuthenticationProvider(clientId, this.defaultUserScopes, options);
```

Comment:
This must be using an old version of the MSALAuthenticationProvider, this is not how it works in v2 of msgraph Javascript SDK.

Reply:
Yes. Coz we are still using the sdk version at 1.5.0

Reply:
I would like this to be updated to the 2.0 SDK.  Also, we fixed the AuthProvider so that you create the UserAgent externally and then pass it into the AuthProvider, so there should be no need for an AuthProvider to wrap the AuthProvider, you should be able to just use the ImplicitMSALAuthProvider directly without having to wrap it and re-implement the AuthenticationProvider interface.